runner/runners/invoke.go

```diff
@@ -368,71 +369,62 @@ func (inv *Invoker) run(cmd *runner.Command, id runner.RunID, abortCh chan struc
+				defer func() {
+					uploadTimer.Stop()
+				}()
+
+				// generate a unique id that's appended to the job id to create a unique identifier for logs
+				logUid, _ := uuid.NewV4()
+				stdlogName := "stdlog"
+				stderrName := "stderr"
+				stdoutName := "stdout"
+
+				var isAborted bool
+				// upload stdlog
+				logId := fmt.Sprintf("%s_%s/%s", cmd.JobID, logUid, stdlogName)
+				_, isAborted = inv.uploadLog(logId, stdlog.AsFile(), abortCh)
+				if isAborted {
+					return runner.AbortStatus(id, tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
 
-			ingestCh := make(chan interface{})
-			go func() {
-				snapshotID, err := inv.filerMap[runType].Filer.Ingest(tmp)
-				if err != nil {
-					ingestCh <- err
-				} else {
-					ingestCh <- snapshotID
 				}
-			}()
 
-			var snapshotID string
-			select {
-			case <-abortCh:
-				if err := inv.filerMap[runType].Filer.CancelIngest(); err != nil {
-					log.Errorf("Error canceling ingest: %s", err)
+				// upload stderr
+				// TODO: remove when we transition to using only stdlog in run status
+				logId = fmt.Sprintf("%s_%s/%s", cmd.JobID, logUid, stderrName)
+				stderrUrl, isAborted = inv.uploadLog(logId, stderr.AsFile(), abortCh)
+				if isAborted {
+					return runner.AbortStatus(id, tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
 				}
-				// Cancel call above should cause ingest to exit sooner, but still wait for return to release worker
-				<-ingestCh
-				return runner.AbortStatus(id, tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-			case res := <-ingestCh:
-				switch res.(type) {
-				case error:
-					log.WithFields(
-						log.Fields{
-							"tag":    cmd.Tag,
-							"jobID":  cmd.JobID,
-							"taskID": cmd.TaskID,
-						}).Errorf("Error ingesting results: %v", res)
-				default:
-					snapshotID = res.(string)
+
+				// upload stdout
+				// TODO: remove when we transition to using only stdlog in run status
+				logId = fmt.Sprintf("%s_%s/%s", cmd.JobID, logUid, stdoutName)
+				stdoutUrl, isAborted = inv.uploadLog(logId, stdout.AsFile(), abortCh)
+				if isAborted {
+					return runner.AbortStatus(id, tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
 				}
+			} else {
+				log.Infof("log uploader not initialized, skipping logs upload to storage")
 			}
-			rts.outputEnd = stamp()
-			rts.invokeEnd = stamp()
-
-			// Note: only modifying stdout/stderr refs when we're actively working with snapshotID.
-			status := runner.CompleteStatus(id, snapshotID, st.ExitCode,
+			status := runner.CompleteStatus(id, "", st.ExitCode,
 				tags.LogTags{JobID: cmd.JobID, TaskID: cmd.TaskID, Tag: cmd.Tag})
-			if cmd.SnapshotID != "" {
-				status.StdoutRef = snapshotID + "/" + stdoutName
-				status.StderrRef = snapshotID + "/" + stderrName
+
+			// Note: only modify stdout/stderr refs when logs are successfully uploaded to storage
+			if stderrUrl != "" {
```

Comment:
Nit but these ifs are unnecessary if following std golang convention of setting these to "" in the error case (which we are)

Reply:
yep, removed!